src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -204,10 +204,10 @@ public DiagnosticBuilderInternal(TextSpan textSpan)
                 "BCP036",
                 $"The property '{property}' expected a value of type {expectedType} but the provided value is of type {actualType}.");
 
-            public ErrorDiagnostic DisallowedProperty(object property, object type) => new ErrorDiagnostic(
+            public ErrorDiagnostic DisallowedProperty(object property, object type, IEnumerable<string> validUnspecifiedProperties) => new ErrorDiagnostic(
                 TextSpan,
                 "BCP037",
-                $"The property '{property}' is not allowed on objects of type {type}.");
+                $"The property '{property}' is not allowed on objects of type {type}." + (validUnspecifiedProperties.Any() ? $" Permissible properties include '{validUnspecifiedProperties.ConcatString("', '")}'." : ""));
```

Comment:
>$"The property '{property}' is not allowed on objects of type {type}." + (validUnspecifiedProperties.Any() ? $" Permissible properties include '{validUnspecifiedProperties.ConcatString("', '")}'." : "")) [](start = 16, length = 203)

Should probably be a separate string to make loc easier later. We could also make it a separate error code too. (BCP038 is available :))

Reply:
Good point.